[Bug]: adguardhome — missing __main__ guard makes the _resolve_cli() dev fallback a no-op

### Affected Software/Harness

adguardhome

### Version / Commit

6f372d3 (main, 2026-08-13)

### Operating System

Linux

### Python Version

3.13.5

### Steps to Reproduce

Note upfront: this reproduces on a plain checkout **without** `pip install -e .`. With the harness
installed, the console script is on `PATH` and everything below passes — which is likely why it has
gone unnoticed.

```bash
cd adguardhome/agent-harness
# no `cli-anything-adguardhome` on PATH, no ~/.config/cli-anything-adguardhome.json present
env -u AGH_HOST -u AGH_PORT -u AGH_USERNAME -u AGH_PASSWORD PYTHONPATH=. \
  python3 -m pytest cli_anything/adguardhome/tests/test_full_e2e.py::TestCLISubprocess -v -s
```

`-s` surfaces the `[_resolve_cli]` line so it is visible which path was taken; unsetting `AGH_*` and
checking for an existing config file matters because `_run()` passes the full environment through and
`load_config()` reads both. Selecting the class keeps the run away from the `TestDockerE2E` fixture,
which starts by executing `docker rm -f agh-cli-test`.

### Expected Behavior

`_resolve_cli()` in `cli_anything/adguardhome/tests/test_full_e2e.py` falls back to
`python -m cli_anything.adguardhome.adguardhome_cli` when the console script is not on `PATH`. I
expected that fallback to invoke the CLI, so the subprocess tests either pass or fail on real
assertions.

### Actual Behavior

The fallback module produces no output at all. `cli_anything/adguardhome/adguardhome_cli.py` defines
`main()` (line 86) but never calls it under `if __name__ == "__main__":`, so running it as a module
executes the decorators and exits 0 with empty stdout:

```console
$ PYTHONPATH=. python3 -m cli_anything.adguardhome.adguardhome_cli --help ; echo "rc=$?"
rc=0                                   # no output at all

$ PYTHONPATH=. python3 -m cli_anything.adguardhome --help
Usage: python -m cli_anything.adguardhome [OPTIONS] COMMAND [ARGS]...
```

(The package path works because `__main__.py` has the guard.)

Five of the seven subprocess tests fail as a result:

```
FAILED TestCLISubprocess::test_help
FAILED TestCLISubprocess::test_config_show_json
FAILED TestCLISubprocess::test_config_show_default_host
FAILED TestCLISubprocess::test_help_subcommands_listed
FAILED TestCLISubprocess::test_filter_help
```

adguardhome looks like an outlier here rather than an intended design: of the 69 `*_cli.py` files
under `cli_anything/` in this repo, 62 have the `__main__` guard. The seven without it are
adguardhome, eth2-quickstart, firefly-iii, mubu, quietshrink, siyuan and zotero — so the same fix
may apply to those as well.

**Suggested fix** — append to `adguardhome_cli.py`:

```python
if __name__ == "__main__":
    main()
```

### Relevant Logs / Tracebacks

```shell
# plain checkout, no install, TestCLISubprocess only
$ env -u AGH_HOST -u AGH_PORT -u AGH_USERNAME -u AGH_PASSWORD PYTHONPATH=. \
    python3 -m pytest cli_anything/adguardhome/tests/test_full_e2e.py::TestCLISubprocess -q -s
[_resolve_cli] Falling back to: /usr/bin/python3 -m cli_anything.adguardhome.adguardhome_cli
5 failed, 2 passed in 2.78s

# whole test directory, same conditions (adds 24 unit tests and 5 skipped Docker tests)
$ PYTHONPATH=. python3 -m pytest cli_anything/adguardhome/tests/ -q
5 failed, 26 passed, 5 skipped in 2.43s

# the five failures, in detail
E   json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
E   AssertionError: assert 'filter' in ''
E    +  where '' = CompletedProcess(args=['/usr/bin/python3', '-m',
     'cli_anything.adguardhome.adguardhome_cli', '--help'], returncode=0, stdout='', stderr='').stdout

# same non-Docker test set in a clean venv after `pip install -e .` (no Docker access here)
$ CLI_ANYTHING_FORCE_INSTALLED=1 python -m pytest cli_anything/adguardhome/tests/ \
    --deselect cli_anything/adguardhome/tests/test_full_e2e.py::TestDockerE2E
31 passed, 5 deselected in 2.22s
```

### Additional Context

Two of the subprocess tests cannot catch this on their own: `test_rewrite_help`
(`test_full_e2e.py:186`) and `test_blocking_help` (`test_full_e2e.py:190`) assert only
`result.returncode == 0`, so they stayed green while the CLI printed nothing. Their sibling tests
(`test_filter_help`, `test_help`) additionally check stdout. Adding one matching `assert "..." in
result.stdout` line to each would make the pair fail loudly in this situation. Purely a suggestion —
the minimal form matches the example in `HARNESS.md`, so this may well be deliberate.

One open question while you are in this file: the `TestDockerE2E` fixture calls `pytest.skip(...)`
when Docker is unavailable. Is that the intended behaviour, given the "no graceful degradation" rule
for backend-dependent tests? Happy to follow whichever way you decide.

I am glad to open a PR with the guard (and the same one-liner for the other six harnesses) if that
is useful.

--- a/adguardhome/agent-harness/cli_anything/adguardhome/adguardhome_cli.py
+++ b/adguardhome/agent-harness/cli_anything/adguardhome/adguardhome_cli.py
@@ -674,3 +674,7 @@ def tls_(ctx: click.Context):
 def tls_status(ctx: click.Context):
     client = make_client(ctx)
     output(server_core.get_tls_status(client), ctx.obj["as_json"])
+
+
+if __name__ == "__main__":
+    main()

--- a/adguardhome/agent-harness/cli_anything/adguardhome/tests/test_full_e2e.py
+++ b/adguardhome/agent-harness/cli_anything/adguardhome/tests/test_full_e2e.py
@@ -186,10 +186,12 @@ class TestCLISubprocess:
     def test_rewrite_help(self):
         result = self._run(["rewrite", "--help"])
         assert result.returncode == 0
+        assert "list" in result.stdout
 
     def test_blocking_help(self):
         result = self._run(["blocking", "--help"])
         assert result.returncode == 0
+        assert "parental" in result.stdout
 
 
 # ---------------------------------------------------------------------------

--- a/adguardhome/agent-harness/setup.py
+++ b/adguardhome/agent-harness/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup, find_namespace_packages
 
 setup(
     name="cli-anything-adguardhome",
-    version="1.0.0",
+    version="1.0.1",
     description="CLI harness for AdGuardHome - control your ad blocker from the command line",
     packages=find_namespace_packages(include=["cli_anything.*"]),
     install_requires=[

--- a/registry.json
+++ b/registry.json
@@ -122,7 +122,7 @@
     {
       "name": "adguardhome",
       "display_name": "AdGuardHome",
-      "version": "1.0.0",
+      "version": "1.0.1",
       "description": "DNS ad-blocking and network infrastructure management via AdGuardHome REST API",
       "requires": "AdGuardHome instance running",
       "homepage": "https://adguard.com/adguard-home/overview.html",

